File: novel-backend/novel-service/src/main/java/com/novel/controller/front/NewsController.java
package com.novel.controller.front;

import com.novel.result.Result;
import com.novel.service.NewService;
import com.novel.user.dto.news.LaetstNewsView;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@Tag(name = "新闻模块")
@RequestMapping("/news")
@RestController
public class NewsController {
	
	@Autowired
	private NewService newService;
	@GetMapping("latest")
	public Result<List<LaetstNewsView>> latestList() {
		var list = newService.latestList();
		return Result.ok(list);
	}
}

File: novel-admin-react/src/__generated/services/NewsController.ts
import type {Executor} from '../';
import type {LaetstNewsView, Result} from '../model/static/';

export class NewsController {
    
    constructor(private executor: Executor) {}
    
    readonly latestList: () => Promise<
        Result<ReadonlyArray<LaetstNewsView>>
    > = async() => {
        let _uri = '/news/latest';
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<Result<ReadonlyArray<LaetstNewsView>>>;
    }
}

export type NewsControllerOptions = {
    'latestList': {}
}
